### DISCUSSION TITLE
The lack of 'pick up items from all nearby tiles' is really bugging me.

### DISCUSSION THREAD

**Discussion by Klokinator** [cataclysmbn/Cataclysm-BN#2658] [General]
I've been playing vanilla CDDA for a while now. I set up a bunch of keybinds by basically overhauling the menu, then someone told me about Bright Nights. I mostly love this mod and will probably semi-permanently switch to it... but there's a big problem.

The 'pickup items' buttons in Bright Nights only have two settings. "Pick up items at a specific tile you have to select via highlight" and "Pick up items directly beneath your feet."

In vanilla CDDA, I have "Pick up items from all nearby tiles" bound to 'e' but the equivalent option isn't in Bright Nights.

Could this be added? I use it constantly and it's more convenient than the advanced inventory manager (which I mostly use for transferring/consolidating large quantities of items or dropping them, and which doesn't allow me to inspect/extract items _inside_ other items.)

**Comment by scarf005** [comment]
sure, it'd be great if you could point to the PR that added said feature in DDA, which would make the porting easier.

**Comment by Klokinator** [comment]
Unfortunately, I have no idea which PR that would be. I know how Github works, but I know nothing about CDDA's code or how I'd even look for it. I don't even know what sort of specific function to look up.

For all I know, the function has been in the game for years; I've only been playing CDDA for about a month, so I'm not familiar with the code development over time. It could have been added a full year ago for all I know.

**Comment by Klokinator** [comment]
Oh, a bit of searching did bring me to this. Maybe it helps?

https://github.com/CleverRaven/Cataclysm-DDA/pull/52683

**Reply by scarf005** [reply]
yeah, it'd help a lot.

**Comment by Klokinator** [comment]
Here are the two I've found.
https://github.com/CleverRaven/Cataclysm-DDA/pull/51033
https://github.com/CleverRaven/Cataclysm-DDA/pull/52683

I think these should cover the functionality, based on what I've read. I'm not sure if later PRs added additional functionality I'm unaware of.

**Comment by Klokinator** [comment]
@scarf005 Were those PRs adequate or should I see if there are other relevant ones?

**Reply by scarf005** [reply]
yeah, i'll take a look when i get time.

**Reply by scarf005** [reply]
the problem is DDA's pickup_all feature is interconnected to its new inventory preset, which wouldn't be an easy port.

**Reply by Klokinator** [reply]
> the problem is DDA's pickup_all feature is interconnected to its new inventory preset, which wouldn't be an easy port.

That's unfortunate. Well, if you do port it, many thanks from me. If not, I'll understand. Thanks for the prompt responses!
